packages/next/src/server/lib/router-utils/filesystem.ts

```diff
@@ -421,12 +421,17 @@ export async function setupFsCheck(opts: {
         return lruResult
       }
 
-      const { basePath } = opts.config
+      const { basePath, assetPrefix } = opts.config
 
-      if (basePath && !pathHasPrefix(itemPath, basePath)) {
+      if (
```

Comment:
Instead of checking both each time we should have a source of truth value that is either the `basePath` or `assetPrefix`.

Reply:
The problem with that is that if both a basePath and an assetPrefix are set, it could start from either one;
- bundles will be fetched with itemPath: `/custom-asset-prefix/_next/static/.....`
- pages with itemPath: `/custom-base-path/...`

So something like 
```ts
const prefix = assetPrefix || basePath;

if (prefix && !pathHasPrefix(itemPath, prefix)) {
  return null
}
itemPath = removePathPrefix(itemPath, prefix) || '/'
```
would result in a 404 for page, instead of bundles (The other way around to the current issue)

Reply:
Hmm I see, it does look like we allow these values to diverge so keeping the checks separate like this seems alright.